openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,201 @@
+                    self.USERNAME_SUFFIX_LENGTH) if duplicated_username else user.username,
+                "users[0][password]": self._get_random_string(settings.ICC_HASH_LENGTH),
+                "users[0][firstname]": user.first_name,
+                "users[0][lastname]": user.last_name,
+                "users[0][email]": user.email,
+                "users[0][auth]": "saml2",
+            }
+
+            response = requests.post(
+                url=settings.ICC_BASE_URL,
+                data=request_data,
+            )
+            log_details["request_payload"] = request_data
+        except Exception as error:  # pylint: disable=broad-except
+            log_details["request_payload"]["wstoken"] = "icc-api-token"
+            log_details["response"] = {"error": "Failed to create ICC user. Reason: %s" % str(error)}
+
+            LOG.error("Failed to create ICC user. Reason: %s", str(error))
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
+        else:
+            icc_user = self._get_icc_user_from_xml_response(response, "create_user")
+
+        return icc_user
+
+    def _validate_icc_user(self, data, icc_user):
+        """
+        Ensure the user creation avoiding failure for username duplicates.
+        """
+        duplicated_username = False
+        call_counter = 0
+
+        while not icc_user and call_counter <= self.USER_CREATION_MAX_RETRIES:
+            icc_user = self._create_icc_user(data, duplicated_username)
+            duplicated_username = True
+            call_counter += 1
+
+        return icc_user
+
+    def _get_icc_user_from_xml_response(self, response, method_type):
+        """
+        Method that receives a xml response and convert it to json object and returns it.
+        """
+        icc_user = {}
+        OBJECT_POSITION = 0
+
+        try:
+            response = xmltodict.parse(response.content)
+            fields = (response["RESPONSE"]["MULTIPLE"]["SINGLE"]["KEY"] if method_type == "create_user" else
+                      response["RESPONSE"]["SINGLE"]["KEY"][OBJECT_POSITION]["MULTIPLE"]["SINGLE"]["KEY"])
+
+            for field in fields:
+                if field.get("@name") == "id":
+                    icc_user["id"] = field.get("VALUE")
+                if field.get("@name") == "username":
+                    icc_user["username"] = field.get("VALUE")
+                    break
+        except Exception:  # pylint: disable=broad-except
+            pass
+
+        return icc_user
+
+    def _get_random_string(self, length):
+        """
+        Method that generates and return a random string.
+        """
+        random_string = (
+            uuid4().hex[:length] if length == self.USERNAME_SUFFIX_LENGTH else configuration_helpers.get_value(
+                'DEFAULT_USER_TESTING_PASSWORD'))
```

Comment:
```suggestion
        random_string = (
            uuid4().hex[:length] if length == self.USERNAME_SUFFIX_LENGTH else
            configuration_helpers.get_value('DEFAULT_USER_TESTING_PASSWORD')
        )
```
Minor suggestion. @ivanvgh

Reply:
Suggestion added.